### DISCUSSION TITLE
Average of metric value over a time window

### DISCUSSION THREAD

**Discussion by prasad-pillala-by** [kedacore/keda#993] [General]
Hi,

My requirement is to autoscale based on a average of kafka lag for last 10 minutes. For example, if the average of incoming kafka messages stays above 1000 for last 10 minutes, then I want to add more pods using HPA. 

Means, I want to avoid acting on a sudden spike of 2000 messages 1 time, which can be handled in 5 minutes by current pod. If in-flow continues for 10 minutes and the average is more than 1000; then I want to add more pods to handle the load.

Can you please help me on how to achieve this?

Thanks,
Prasad

**Comment by tomkerkhove** [comment]
Looks like you are looking for [`pollingInterval`](https://keda.sh/docs/1.5/concepts/scaling-deployments/#details):

> This is the interval to check each trigger on. By default KEDA will check each trigger source on every ScaledObject every 30 seconds.
> 
> Example: in a queue scenario, KEDA will check the queueLength every pollingInterval, and scale the deployment up or down accordingly.

**Reply by prasad-pillala-by** [reply]
I think pollingInterval check the instant value at this moment. Does it support average of last 10 minutes values i.e. 20 values average?

**Reply by tomkerkhove** [reply]
Oh that way, I don't think that's supported at the moment, right @zroubalik

**Reply by prasad-pillala-by** [reply]
If this is not available today. Can you consider it as an enhancement?
Thanks

**Reply by zroubalik** [reply]
Yeah, it is not available. @prasad-by let's open an issue with feature request. So we can discuss it there.
